packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -906,6 +910,30 @@ export function TransferPanelMain({
                     forToken={selectedToken}
                     prefix={selectedToken ? 'BALANCE: ' : ''}
                   />
+                  {/* In deposit mode, when user selected USDC on mainnet,
+                  the UI shows the Arb One balance of both USDC.e and native USDC */}
+                  {app.isDepositMode &&
+                    ((isArbitrumOne &&
+                      isTokenMainnetUSDC(selectedToken?.address)) ||
+                      (isArbitrumGoerli &&
```

Comment:
The show logic seems to be the same as governed by the variable - `showUSDCNotice`. We should reuse this variable or rename it to something generic like `showAdditionalUSDCData`. In any case - having such a complex condition in JSX doesn't look the best.

Reply:
it's just a bit verbose, i wouldn't call the logic complex as it's rather straightforward

i'd refrain from batching the conditions for different USDC-related UI as it is a special case and might change anytime